src/Symfony/Component/Messenger/Bridge/AmazonSqs/Transport/Connection.php

```diff
@@ -167,66 +170,84 @@ private function getPendingMessages(): \Generator
     private function getNewMessages(): \Generator
     {
         if (null === $this->currentResponse) {
-            $this->currentResponse = $this->request($this->getQueueUrl(), [
-                'Action' => 'ReceiveMessage',
+
+            $this->currentResponse = $this->client->receiveMessage([
+                'QueueUrl' => $this->getQueueUrl(),
                 'VisibilityTimeout' => $this->configuration['visibility_timeout'],
                 'MaxNumberOfMessages' => $this->configuration['buffer_size'],
                 'WaitTimeSeconds' => $this->configuration['wait_time'],
             ]);
         }
 
-        if ($this->client->stream($this->currentResponse, $this->configuration['poll_timeout'])->current()->isTimeout()) {
+        if (!$this->fetchMessage()) {
             return;
         }
 
-        $xml = new \SimpleXMLElement($this->currentResponse->getContent());
-        foreach ($xml->ReceiveMessageResult->Message as $xmlMessage) {
+        yield from $this->getPendingMessages();
+    }
+
+    private function fetchMessage(): bool
+    {
+        if (!$this->currentResponse->resolve($this->configuration['poll_timeout'])) {
+            return false;
+        }
+
+        foreach ($this->currentResponse->getMessages() as $message) {
             $this->buffer[] = [
-                'id' => (string) $xmlMessage->ReceiptHandle,
-            ] + json_decode($xmlMessage->Body, true);
+                    'id' => $message->getReceiptHandle(),
+                ] + json_decode($message->getBody(), true);
         }
 
         $this->currentResponse = null;
 
-        yield from $this->getPendingMessages();
+        return true;
     }
 
     public function setup(): void
     {
-        $this->call($this->configuration['endpoint'], [
-            'Action' => 'CreateQueue',
```

Comment:
That were a bug. When user provides an external accountId, we shouldn't try to create the queue on the Owner's account